File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/templatemethod/transport/Bike.java
// Bike.java

package com.bigboxcode.designpattern.templatemethod.transport;

public class Bike extends Transport {
    @Override
    void createBody() {
        System.out.println("Creating Bike Body");
    }

    @Override
    void addEngine() {
        System.out.println("Adding Engine to Bike");
    }

    @Override
    void addWheel() {
        System.out.println("Adding 2 Wheels to Bike");
    }

    @Override
    void addWing() {
        // Not required for Bike
    }
}

File: design-patterns/php/templatemethod/transport/Bike.php
<?php
// Bike.php

namespace BigBoxCode\DesignPattern\TemplateMethod\Transport;

class Bike extends Transport {
    public function createBody(): void {
        echo "Creating Bike Body\n";
    }

    public function addEngine(): void {
        echo "Adding Engine to Bike\n";
    }

    public function addWheel(): void {
        echo "Adding 2 Wheels to Bike\n";
    }

    public function addWing(): void {
        // not required for Bike
    }
}